frontend/src/app/services/dataset.service.spec.ts

```diff
@@ -0,0 +1,66 @@
+import {async, inject, TestBed} from '@angular/core/testing';
+import {HttpClientModule, HttpRequest} from '@angular/common/http';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
+import {environment} from '../../environments/environment';
+import {DatasetService} from './dataset.service';
+import {Dataset} from '../model/ScanMetadata';
+import {API_URL} from '../utils/ApiUrl';
+
+describe('Service: DatasetService', () => {
+    const EXAMPLE_DATASETS = [
+        {
+            name: 'DATASET_NAME_1',
+            key: 'DATSET_KEY_1'
+        },
+        {
+            name: 'DATASET_NAME_2',
+            key: 'DATSET_KEY_2'
+        }];
+
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [
+                HttpClientModule,
+                HttpClientTestingModule
+            ],
+            providers: [
+                DatasetService
+            ]
+        });
+    });
+
+    it('should get available datasets', async(inject([DatasetService, HttpTestingController],
+        (service: DatasetService, backend: HttpTestingController) => {
+
+            service.getAvailableDatasets().then((results: Dataset[]) => {
```

Comment:
I'm having problems with understanding this tests.  There are no explicit differences apart from the fact that in the second test we expect error to happen. And If I understood correctly it is based on the fact that we are not authorized, right? Are we passing the token somewhere?

If the second test is for a case when user is not logged in maybe the title should be `shouldn't get available datasets when not logged in`.

Reply:
We are just testing the behaviour of service method in case of error http response (here response values doesn't matter).